packages/hs-react-ui/src/components/Table/Table.tsx

```diff
@@ -1,48 +1,83 @@
 import React, { useState, useEffect } from 'react';
 import styled, { StyledComponentBase } from 'styled-components';
 import useResizeObserver from 'use-resize-observer';
+import { mdiArrowDown, mdiChevronRight, mdiChevronDown, mdiChevronUp } from '@mdi/js';
 import Icon from '@mdi/react';
-import { mdiArrowDown } from '@mdi/js';
-
 import colors from '../../enums/colors';
 import fonts from '../../enums/fonts';
 
+/** Types and Interfaces */
+
+export type CellOptions = {
+  RenderedCell: any;
+  headerColumnKey: string; //
+  breakPointHit: boolean;
+  row: columnTypes;
+  index: number;
+  indexModifier?: number;
+  CollapseIcon?: any;
+  groupIndex?: number;
+  isCollapsed?: boolean;
+  cIcon?: any;
+  eIcon?: any;
+  groupLabelDataString?: string;
+};
+
+export type ExpansionIconProps = {
+  collapsedIcon: string;
+  expandedIcon: string;
+  isCollapsed: boolean;
+  groupHeaderPosition: 'above' | 'below';
+  onClick: any;
+};
+
 export interface columnTypes {
   [index: string]: {
     name?: string;
     width?: string;
     minTableWidth?: number;
     sortable?: boolean;
     sortFunction?: Function;
-
+    isGroupLabel?: boolean;
     cellComponent?: any;
     rowComponent?: any;
     headerCellComponent?: any;
+    groupCellComponent?: any;
   };
 }
 
 export type TableProps = {
+  isCollapsable?: boolean;
   columnGap?: string;
-  defaultSort?: [string, boolean]; // key, direction
-  data?: columnTypes[];
   columns: columnTypes;
-
+  data?: columnTypes[] | Array<Array<columnTypes>>;
+  defaultSort?: [string, boolean]; // key, direction
+  groupHeaderPosition?: 'above' | 'below';
+  expansionIconComponent?: React.FunctionComponent<ExpansionIconProps>;
+  collapsedIcon?: string | React.FunctionComponent<any>;
+  expandedIcon?: string | React.FunctionComponent<any>;
```

Comment:
Should we just allow anything to be passed in as an icon and then pass it the data it needs to render (isCollapsed)

Reply:
I think that's a good idea. I've been trying to get as much customizability as I can and in doing so I think I'm going a little overboard. I was going for these use cases:

- Use the default icons that we have implemented (`mdiChevronDown` & `mdiChevronRight`)
- Use the default Icon component with different material ui components by using only `collapsedIcon` or `expandedIcon` as string props
- Use a completely expansion icon by using `expansionIconComponent` - completely replacing the material Icon we're using (`collapsedIcon` and `expandedIcon` are not required in this situation)
- Use a completely custom expansion icon component that also icon overrides. The use case I was thinking of here is an application that uses a custom expansion icon component that they've defined and can pass into any table, but they may want to customize the icon based on props instead of creating a whole new icon component for version. Essentially their own custom version of how our `ExpansionIcon` is implemented, but not requiring material-ui

I'd really like some input on this, can we talk about the approach and hash out what would be best and easiest for consumption without over complicating things?

Reply:
Ideally we should have one prop `ExpansionIcon`. We pass that component the "isCollapsed" info so it can decide how it wants to look/what icon it wants to return.